src/store/flow/block/choice.ts

```diff
@@ -113,20 +114,30 @@ export const actions: ActionTree<IEmptyState, IRootState> = {
    */
   choice_updateFirstSynonymForActiveLanguages(
     {getters, rootGetters, dispatch, commit},
-    {blockId, resourceId, value}: { blockId: IBlock['uuid'], resourceId: IResource['uuid'], value: string },
+    {blockId, resourceId, value, oldValue}: { blockId: IBlock['uuid'], resourceId: IResource['uuid'], value: string, oldValue: string },
   ) {
     const flow: IFlow = rootGetters['flow/activeFlow']
     const languages = flow.languages
 
     languages.forEach(({id: language}) => {
-      // TODO Don't replace if existing value was changed by the user
-      dispatch('choice_setSynonymForLanguage', {
-        blockId,
-        resourceId,
-        language,
-        index: 0,
-        value,
-      })
+      const choice = getters.choice_findChoice(blockId, resourceId) as IChoice
+      const existingValue = choice.text_tests?.find(synonym => synonym.language === language)?.test_expression
+
+      const isNew = existingValue === undefined
+      const hasValueToMatch = oldValue !== undefined
+      const hasMatchingValues = hasValueToMatch && existingValue === textValueToExpression(oldValue)
+      const shouldUpdateValue = isNew || !hasValueToMatch || hasMatchingValues
+
+      // Make sure the 1st synonym has not been changed by the user
+      if (shouldUpdateValue) {
```

Comment:
I thought we may need to use vendor_metadata for this part, but I'm glad there is an alternative way, thanks